shard.go

```diff
@@ -10,19 +10,25 @@ import (
 
 type onRemoveCallback func(wrappedEntry []byte, reason RemoveReason)
 
+type metadata struct {
+	requestCount uint32
+}
+
 type cacheShard struct {
 	hashmap     map[uint64]uint32
 	entries     queue.BytesQueue
 	lock        sync.RWMutex
 	entryBuffer []byte
 	onRemove    onRemoveCallback
 
-	isVerbose  bool
-	logger     Logger
-	clock      clock
-	lifeWindow uint64
+	isVerbose    bool
+	statsEnabled bool
+	logger       Logger
+	clock        clock
+	lifeWindow   uint64
 
-	stats Stats
+	hashmapStats map[uint64]*uint32
```

Comment:
`uint32` should be enough here

Reply:
I used *uint32 here because you can't do `atomic.AddUint32(&s.hashmapStats[key], 1)` with a map

Reply:
We don't need atomic at all, 'cause all the write happens under a mutex.